rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,406 @@
+When adding a trip, there are three kinds of trips which can be added:
+- `"arrival"`: a trip arriving to `endLocation` at `time`
+- `"departure"`: a trip departing from `startLocation` at `time`
+
+## Events
+### Operator signed in
+At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They do this by either physically signing a paper trainsheet, or by typing their badge number into Glides. This event represent either type of signin, but may not be generated in the case where an operator only signs into the paper trainsheet.
+
+Event type: `com.mbta.ctd.glides.operator-signed-in`
+Fields in the event:
+- `author` (Author): the inspector who signed in the operator
+- `operator` (Operator): the operator who signed in
+- `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
+
+### Trip added
+This creates a new trip in the timesheet. It may or may not be a new trip relative to GTFS (see the splitting a train reference example).
+
+Event type: `com.mbta.ctd.glides.trip-added`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): provides a unique identifier for the newly created trip
+- `route` (string, optional): if specified, the route tag for the trip. This is different from a GTFS route ID.
+- `startLocation` (Location, conditionally required): where the trip will be starting. Required if `tripType` is `"departure"`.
+- `endLocation` (Location, conditionally required): where the trip will be ending. Required if `tripType` is `"arrival"`.
+- `time` (Time, conditionally required): when the added trip is expected to depart (`tripType` of `"departure"`) or arrive (`tripType` of `"arrival"`). Required when `tripType` is `"departure"` or if `previousTripKey` is not specified.
+- `tripType` (TripType): whether `time` for the new trip is the when the trip is departing `startLocation` (`"departure"`) or when the trip is arriving at `endLocation` (`"arrival"`).
+- `previousTripKey` (Trip Key, optional): links the newly created trip to the trip immediately before it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message.
+- `nextTripKey` (Trip Key, optional): links the newly created trip to the trip after it. If the Trip Key refers to an added trip, it may not have been see in the event stream at the time of this message.
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+- `reason` (string, optional): text description about why the trip is being added
```

Comment:
In Glides, we don't ascribe any specific meaning to the `comment` field. It's not really meant to provide a reason for a specific edit (although it could), it's more to describe the whole trip. The comment also persists with the trip and might be updated over time, instead of being ephemeral like the edit event.

If this field is meant to correspond to the Glides `comment` field, then I'd suggest renaming it to `comment`. If not, I'm not sure how Glides should decide what to show here, and we probably want to add a `comment` field.

(applies to both this and the `trip-updated` reason field, but does not apply to the dropped trip reason. That one really is a reason for the edit.)

Reply:
Gotcha; I didn't realize that the reason for the dropped trip was a separate field, but I see it now in the UI. I'll revert this change and leave it as an open question for later.

Reply:
I also added `comment` as a field on the trip.